subsys/settings/src/settings_store.c

```diff
@@ -118,9 +118,9 @@ int settings_save_one(const char *name, char *value)
  * Walk through all registered subsystems, and ask them to export their
  * config variables. Persist these settings.
  */
-static void settings_store_one(char *name, char *value)
+static int settings_store_one(char *name, char *value)
 {
-	settings_save_one(name, value);
+	return settings_save_one(name, value);
```

Comment:
Now that the prototypes match, maybe get rid of `settings_store_one()` and pass `settings_save_one` to whatever uses this as a callback?